playbooks/roles/edxapp/templates/edx/app/edxapp/worker.sh.j2

```diff
@@ -4,19 +4,33 @@
 
 {% set edxapp_venv_bin = edxapp_venv_dir + "/bin" %}
 source {{ edxapp_app_dir }}/edxapp_env
+{% set executable = edxapp_venv_bin + '/celery' %}
+
 {% if COMMON_ENABLE_NEWRELIC_APP %}
-{% set executable = edxapp_venv_bin + '/newrelic-admin run-program ' + edxapp_venv_bin + '/celery' %}
+{% set executable = edxapp_venv_bin + '/newrelic-admin run-program ' + executable %}
 
 export NEW_RELIC_CONFIG_FILE="{{ edxapp_app_dir }}/newrelic.ini"
 if command -v ec2metadata >/dev/null 2>&1; then
   INSTANCEID=$(ec2metadata --instance-id);
   HOSTNAME=$(hostname)
   export NEW_RELIC_PROCESS_HOST_DISPLAY_NAME="$HOSTNAME-$INSTANCEID"
 fi
-{% else %}
-{% set executable = edxapp_venv_bin + '/celery' %}
 {% endif %}
 
+{% if COMMON_ENABLE_DATADOG and COMMON_ENABLE_DATADOG_APP %}
+{% set executable = edxapp_venv_bin + '/ddtrace-run ' + executable %}
+export DD_TAGS="$1"
+# remove the param dd_tags as it is not needed in later execution.
+shift
```

Comment:
It makes me nervous to have this argument be handled specially and stripped off like this. Is there some other way we can pass the info in, e.g. setting a `SERVICE_VARIANT` environment variable in the conf files or (even better) accessing the variant in a template variable from here directly?

Reply:
It looks like we might actually have what we need already -- the conf file has `SERVICE_VARIANT={{ w.service_variant }}` (line 4) and se we should be able to use that when setting DD_TAGS.